react-native-community/react-native-image-picker
# Resolve memory leak in iOS image picker

## Motivation (required)

Currently, in a ReactNative app that is using a "Native Navigation" solution which uses multiple UIViewControllers, using react-native-image-picker will retain the presenting ViewController, causing a memory leak and undesirable behavior.

In our case, since the ViewController was retained, so were the components, which were dispatching off Redux actions in the background

Additionally, I have moved the UIAlertController code _into_ the main thread. UIKit is marked as not being thread-safe, and should only be used on the main thread. 

## Test Plan (required)

I was unable to find a test suite on the iOS target. 

A good manual QA test would involve the following:
- Present a UIViewController modally ("ViewController B")
- Set a breakpoint on the dealloc method of ViewController B
- Have ViewController B present the image picker options ( simply presenting the AlertController is enough to trigger the retain cycle)
- Dismiss ViewController B
- Observe if the dealloc method is called on ViewController B. If it is called, the viewcontroller has properly deallocated. If it has not deallocated, there is a memory leak still (you can observe this on master)

In a proper unit test, I would recommend using a weak ref to ViewController B, and verifying that it is nil after the image picker is dismissed


## Additional comments:

Thanks for this library. 
I noticed 2 extra files generated by XCode included in this PR. They are not part of the .gitignore. If you would like them removed please let me know
